### PR TITLE
[meson] remove meson

### LINKED ISSUE #12449: [meson] delete meson.build files in favor of bazel
Once the following issues have been addressed, all `meson.build` files can be removed (with minimal regressions):

- [x] #12447
- [x] #11563
- [x] #11133
- [ ] #12445
- [x] #12446
- [x] #12455
- [x] #12515 
- [x] #12531
- [x] #11792
- [x] #12562 to ensure lowRISC/misc-linters caches Python wheels (partially address #12538)
- [x] #12494 
- [x] #12069
- [x] #12696
- [x] #12655
- [x] #12953

Additionally the following items will need to be fixed after the meson deletion to cleanup some minor regressions:
- [ ] #12448 (temporary solution will work for the time being)
- [ ] #12504 
- [ ] #12497

### PR DESCRIPTION
**Pull request by timothytrippel**
This PR contains three commits that remove meson from the src tree, making Bazel the only supported build system for OpenTitan.

**Commit 1:** removes all `meson.build` files in favor of using Bazel as the OpenTitan build system.

**Commit 2:** removes meson.build templates used by various code autogen tools

**Commit 3:** removes various references to meson from several tools and code comments throughout the repository
    
This fixes #12449.

Note: a follow on PR updates the website documentation to reflect this change.

### CODE REVIEW

File: hw/top_englishbreakfast/util/prepare_sw.py [hunk 1]
```diff
@@ -74,29 +72,27 @@ def main():
...
+    )
+    parser.add_argument('--top',
+                        '-t',
+                        default='englishbreakfast',
+                        type=str,
+                        help='The alternative top to use')
+    parser.add_argument('--bazel',
+                        default=False,
+                        action='store_true',
+                        help='Whether to build with Bazel instead')
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply hw/top_englishbreakfast/util/prepare_sw.py]: Oops good catch! I meant to remove this argument. Fixed.

Lines 8-11 are referring to a bazel solution for multiple tops, not the use of meson vs. bazel in this script. (I.e. lines 8--11 are saying that this script shouldn't exist in the first place as it dirties the tree outside of the build sandbox which is bad practice.)

[the remark this answers is not in the document]
Comment (andreaskurth) [reply hw/top_englishbreakfast/util/prepare_sw.py]: Okay, I see. Thanks for your work! :+1:

**Comment by luismarques** [comment]
If you grep (case-insensitively) for "meson" you still find various places where Meson should be removed. AFAIK, GitHub doesn't allow you to comment on unchanged files so it's not very convenient to reference all of those places here but grepping will find them easily. Things like this:

```
.bazelignore
5:# TODO: once Meson is no longer used, remove the line below.
```

**Comment by timothytrippel** [comment]
> If you grep (case-insensitively) for "meson" you can still find various places where Meson should be removed. AFAIK, GitHub doesn't allow you to comment on unchanged files so it's not very convenient to reference all of those places here but grepping will find them easily. Things like this:
> 
> ```
> .bazelignore
> 5:# TODO: once Meson is no longer used, remove the line below.
> ```

That was the plan all along hehe but looks like I missed some, thanks!

(Additionally some are addressed in the follow PR #13157, and other stragglers have issues tracking them, i.e., #12504 and #11743, and will be cleaned up in follow on PRs if thats alright?)

**Comment by luismarques** [comment]
> (Additionally some are addressed in the follow PR #13157, and other stragglers have issues tracking them, i.e., #12504 and #11743, and will be cleaned up in follow on PRs if thats alright?)

Is it within the scope of this PR to remove `util/get-toolchain.py`? My understanding is that it's not needed with Bazel, I don't know if people have other uses for it.

This one also looks like it might have been overlooked:

```
util/lowrisc_misc-linters/licence-checker/licence-checker.py
152:    (["meson.build", "toolchain.txt", "meson_options.txt"], HASH),  # Meson
```

File: BUILD.bazel [hunk 2]
```diff
@@ -47,9 +47,6 @@ clang_format_check(
         "./**/vendor/**",
         # Rust cargo build dirs
         "./**/target/**",
-        # Meson build dirs
-        "./build-out/**",
-        "./build-bin/**",
```
Comment (cfrantz) [inline BUILD.bazel:52]: Are we still copying stuff to `build-bin` or `build-out` during any of the build flows?  If yes, we should leave this in for now.

Comment (timothytrippel) [reply BUILD.bazel]: Ah yes, good catch. Added back.

**Comment by timothytrippel** [comment]
> Is it within the scope of this PR to remove `util/get-toolchain.py`? My understanding is that it's not needed with Bazel, I don't know if people have other uses for it.

So I think this is still nice to have (and the updated SW docs still reference using some of the tools for things like disassembling and using GDB) as we don't have Bazel flows for things like launching a GDB debug session with the Bazel managed toolchain yet so I get kept this for the time being.

> This one also looks like it might have been overlooked:
> 
> ```
> util/lowrisc_misc-linters/licence-checker/licence-checker.py
> 152:    (["meson.build", "toolchain.txt", "meson_options.txt"], HASH),  # Meson
> ```

Ah, I originally had removed these lines, but then CI failed as it looks like this script is vendored in. So I will need to have a follow on PR in the other repo that stores this tool and one in this repo that re-vendors in the updated script. Tracking this in #13169.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**